judge/admin/interface.py

```diff
@@ -79,11 +79,27 @@ class BlogPostAdmin(VersionAdmin):
     form = BlogPostForm
     date_hierarchy = 'publish_on'
 
+    def has_view_permission(self, request, obj=None):
+        if obj is None:
+            return request.user.has_perm('judge.view_blogpost')
+        return obj.is_editable_by(request.user)
+
     def has_change_permission(self, request, obj=None):
         if obj is None:
             return request.user.has_perm('judge.change_blogpost')
         return obj.is_editable_by(request.user)
 
+    def get_readonly_fields(self, request, obj=None):
+        if not request.user.has_perm('judge.change_post_visibility'):
+            return ['visible']
+        return []
+
+    def get_queryset(self, request):
+        queryset = BlogPost.objects.all()
+        if not request.user.has_perm('judge.edit_all_post'):
+            queryset = queryset.filter(authors=request.user.profile)
```

Comment:
```suggestion
            queryset = queryset.filter(authors=request.profile)
```

Reply:
Did not know request had a profile field, thanks.